vscode-ext/src/StatusBarComponent.ts

```diff
@@ -38,36 +48,156 @@ export class StatusBarComponent {
-            this.status = status;
+            this.serverStatus = status;
             this.render();
         });
+
+        verifier.client.onVerifyResult((document, results) => {
+            this.verifyStatus.insert(document, results);
+            this.render();
+        });
+
+        verifier.context.subscriptions.push(vscode.workspace.onDidCloseTextDocument((document) => {
+            const documentIdentifier: TextDocumentIdentifier = { uri: document.uri.toString() };
+            this.verifyStatus.remove(documentIdentifier);
+            this.render();
+        }));
+
+        verifier.context.subscriptions.push(vscode.window.onDidChangeVisibleTextEditors(() => {
+            this.render();
+        }));
+
+        verifier.context.subscriptions.push(vscode.workspace.onDidOpenTextDocument(() => {
+            this.render();
+        }));
     }
 
     render() {
         if (this.enabled) {
-            switch (this.status) {
+
+            switch (this.serverStatus) {
+                case ServerStatus.NotStarted:
+                    this.view.text = "$(debug-start) Caesar Inactive";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipMenuText = stoppedTooltipMenu;
+                    break;
                 case ServerStatus.Stopped:
-                    this.view.text = "$(debug-stop) Et tu, Brute?";
+                    this.view.text = "$(debug-start) Et tu, Brute?";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipMenuText = stoppedTooltipMenu;
                     break;
                 case ServerStatus.FailedToStart:
                     this.view.text = "$(warning) Failed to start Caesar";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipMenuText = stoppedTooltipMenu;
                     break;
                 case ServerStatus.Starting:
-                    this.view.text = "$(sync~spin) Starting Caesar...";
+                    this.view.text = "$(loading~spin) Starting Caesar...";
+                    this.view.command = "caesar.showOutput"
                     break;
                 case ServerStatus.Ready:
-                    this.view.text = "$(check) Caesar Ready";
+                    this.handleReadyStatus();
+                    this.view.command = "caesar.showOutput";
+                    this.tooltipMenuText = runningTooltipMenu;
                     break;
                 case ServerStatus.Verifying:
                     this.view.text = "$(sync~spin) Verifying...";
-                    break;
-                case ServerStatus.Finished:
-                    this.view.text = "$(check) Verified";
+                    this.view.command = "caesar.showOutput"
                     break;
             }
+
+            this.renderTooltip();
             this.view.show();
         } else {
             this.view.hide();
         }
     }
+
+    private renderTooltip() {
```

Comment:
Maybe add a comment that this renders the tooltip with the text provided in `this.tooltipMenuText`? Alternatively, just add `tooltipMenuText` as a parameter to this function and don't have it as an attribute at all?